pkg/base/helm.go

```diff
@@ -271,3 +274,105 @@ func checkChartForVersion(file *upstreamtypes.UpstreamFile) (string, error) {
 	// if no determination is made, assume v2
 	return "v2", nil
 }
+
+// insert namespace if it's defined in the spec and not already present in the manifests
+func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
+	var updatedBaseFiles []BaseFile
+	if renderOptions.Namespace != "" {
+		chartsPath, err := ioutil.TempDir("", "charts")
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create temp dir")
+		}
+		defer os.RemoveAll(chartsPath)
+
+		resources := map[string]BaseFile{}
+		var kustomizeResources []string
+		var kustomizePatches []kustomizetypes.PatchStrategicMerge
+		foundGVKNamesMap := map[string]bool{}
+		for _, baseFile := range baseFiles {
+			// write temp files for manifests that need a namespace
+			gvk, manifest := GetGVKWithNameAndNs(baseFile.Content, renderOptions.Namespace)
+			if manifest.APIVersion == "" && manifest.Kind == "" && manifest.Metadata.Name == "" {
+				updatedBaseFiles = append(updatedBaseFiles, baseFile)
+				continue // ignore invalid resources
+			}
+
+			if manifest.Metadata.Namespace == "" {
+				splitPath := strings.Split(baseFile.Path, "/")
+				name := splitPath[len(splitPath)-1]
```

Comment:
you can use `filepath.Base` instead to get the file name from the full path.

example: https://play.golang.org/p/DzlCV-HC-r